tools/codegen/gen-config-mapper/provider_spec_model.go

```diff
@@ -0,0 +1,55 @@
+package genconfigmapper
+
+import (
+	"github.com/hashicorp/terraform-plugin-codegen-spec/resource"
+)
+
+type CodeSpecification struct {
+	DataSource       DataSource       `json:"datasource,omitempty"`
+	DataSourcePlural DataSourcePlural `json:"datasourceplural,omitempty"`
+	Resources        Resource         `json:"resource,omitempty"`
+}
+
+type DataSourcePlural struct {
+	Schema *Schema `json:"schema,omitempty"`
+	Name   string  `json:"name"`
+}
+
+type DataSource struct {
+	Schema *Schema `json:"schema,omitempty"`
+	Name   string  `json:"name"`
+}
+
+type Resource struct {
+	Schema *Schema `json:"schema,omitempty"`
+	Name   string  `json:"name"`
+}
+
+type Schema struct {
+	Description         *string `json:"description,omitempty"`
+	MarkdownDescription *string `json:"markdown_description,omitempty"`
+	DeprecationMessage  *string `json:"deprecation_message,omitempty"`
+
+	// Blocks              Blocks  `json:"blocks,omitempty"`
```

Comment:
@AgustinBettati I don't believe we want to support blocks right?

Reply:
agree, we can revisit if we eventually plan to generate for older resources that have blocks in TPF.

Reply:
removed